### PR TITLE
`OptionChoice` Bug Fix

### PR DESCRIPTION
**Pull request by plun1331**
## Summary

~~Adds `input_type` to `SlashCommandGroup` (Kind of adds on to #676)~~ Fixed in https://github.com/Pycord-Development/pycord/commit/f9e58a51d6a64f0891a5c7370bd1cea2610fde30

Fixed a bug in `OptionChoice` where you couldn't provide a false value (such as `0`) to `value`

## Checklist

- [x] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

**Review by krittick — APPROVED** [review]

**Comment by Dorukyum** [comment]
What does this change? Slash command groups don't get inputs

File: discord/commands/commands.py [hunk 1]
```diff
@@ -773,7 +773,7 @@ def __repr__(self):
 class OptionChoice:
     def __init__(self, name: str, value: Optional[Union[str, int, float]] = None):
         self.name = name
-        self.value = value or name
+        self.value = value if value is not None else name
```
Comment (VincentRPS) [inline discord/commands/commands.py:776]: why did you change this?

**Review by VincentRPS — CHANGES_REQUESTED** [review]

File: discord/commands/commands.py (hunk 1 above)
Comment (Dorukyum) [reply discord/commands/commands.py:776]: > Fixed a bug in OptionChoice where you couldn't provide a false value (such as 0) to `value`

**Comment by plun1331** [comment]
> What does this change? Slash command groups don't get inputs

Groups & Subcommands are internally considered options by Discord.

When the library tries to send them it raises an exception similar to the one in the referenced PR, but with `input_type`.

**Comment by Lulalaby** [comment]
> Groups & Subcommands are internally considered options by Discord.
> 
> When the library tries to send them it raises an exception similar to the one in the referenced PR, but with `input_type`.
that-this

File: discord/commands/commands.py [hunk 2]
```diff
@@ -857,6 +857,7 @@ def __init__(
         )
         self.name = name
         self.description = description
+        self.input_type = SlashCommandOptionType.sub_command_group
```
Comment (s0hv) [inline discord/commands/commands.py:860]: I believe this fix is not correct as the problem is super calling the init of `ApplicationCommand` instead of `Option`. The correct fix would be to fix the super call at line 853. One possible fix for this would be to replace the init call with this to correctly call the init of `Option`.
```py
Option.__init__(
    self,
    SlashCommandOptionType.sub_command_group,
    name=name,
    description=description,
)
```
This would correctly set input_type for the class.

There are other solutions as well as can be seen in [this stackoverflow answer](https://stackoverflow.com/a/39415209/6046713).

Comment (plun1331) [reply discord/commands/commands.py:860]: This is just a temporary fix, we were discussing something like this in the `#discussion` channel in discord.gg/pycord

**Review by Middledot — APPROVED** [review]

**Review by Dorukyum — APPROVED** [review]
